From 958b2c6e2f0141e3d9ec67842331d7af97f2e2e5 Mon Sep 17 00:00:00 2001
From: henning <>
Date: Wed, 22 Apr 2026 13:57:58 +0000
Subject: [PATCH 06/32] grow the buffer to hold the sprintf'd rtable id by 8
 bytes so it can hold the full range an int can express. the previous buffer
 was large enough since the rtable id is clamped to 0.. RT_TABLEID_MAX which
 is 255, however, it is a bit much to ask for analyzers - including those in
 compilers - to detect that, 8 extra bytes on the stack cost us effectively
 nothing, and it feels a bit more robust. triggered by bcook's portable diffs,
 ok claudio

---
 src/usr.sbin/ntpd/util.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/usr.sbin/ntpd/util.c b/src/usr.sbin/ntpd/util.c
index 3371a309847..a0b64ee4b92 100644
--- a/src/usr.sbin/ntpd/util.c
+++ b/src/usr.sbin/ntpd/util.c
@@ -1,4 +1,4 @@
-/*	$OpenBSD: util.c,v 1.29 2025/08/20 10:40:21 henning Exp $ */
+/*	$OpenBSD: util.c,v 1.30 2026/04/22 13:57:58 henning Exp $ */
 
 /*
  * Copyright (c) 2004 Alexander Guy <alexander.guy@andern.org>
@@ -149,7 +149,7 @@ d_to_sfp(double d)
 char *
 print_rtable(int r)
 {
-	static char b[11];
+	static char b[19];
 
 	b[0] = 0;
 	if (r > 0)
-- 
2.53.0

